proxy/http/HttpUserAgent.h

```diff
@@ -0,0 +1,110 @@
+/** @file
+
+  A brief file description
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#pragma once
+
+#include "I_IOBuffer.h"
+#include "I_VConnection.h"
+#include "I_VIO.h"
+#include "ProxyTransaction.h"
+
+class HttpSM;
+using HttpSMHandler = int (HttpSM::*)(int, void *);
+
+enum HttpVC_t {
+  HTTP_UNKNOWN = 0,
+  HTTP_UA_VC,
+  HTTP_SERVER_VC,
+  HTTP_TRANSFORM_VC,
+  HTTP_CACHE_READ_VC,
+  HTTP_CACHE_WRITE_VC,
+  HTTP_RAW_SERVER_VC
+};
+
+struct HttpVCTableEntry {
+  VConnection *vc;
+  MIOBuffer *read_buffer;
+  MIOBuffer *write_buffer;
+  VIO *read_vio;
+  VIO *write_vio;
+  HttpSMHandler vc_read_handler;
+  HttpSMHandler vc_write_handler;
+  HttpVC_t vc_type;
+  HttpSM *sm;
+  bool eos;
+  bool in_tunnel;
+};
```

Comment:
These are obviously not just for "UserAgent". Probably need another header file if they can't stay in HttpSM.h.

Reply:
Do you have a name in mind for this header file?

Reply:
I've selected HttpVCTable.h.